my-app/src/components/Grid.tsx

```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+import styled from '@emotion/styled';
+
+type ResponseGridStyle = {
+	default: number;
+	[breakpoint: string]: number;
+};
+
+interface GridProps {
+	children: React.ReactNode;
+	gap?: string;
+	columns?: number | ResponseGridStyle;
+}
+
+const getGridTemplateColumns = (columns: number | ResponseGridStyle) => {
```

Comment:
타입을 동어반복하지 않고도, `GridProps['columns']`로 좀더 직관적으로 타이핑할 수 있습니다. 참고해보세요.

```suggestion
const getGridTemplateColumns = (columns: GridProps['column']) => {
```